src/client/game.cpp

```diff
@@ -1273,9 +1259,8 @@ void Game::shutdown()
 	if (formspec)
 		formspec->quitMenu();
 
-#ifdef HAVE_TOUCHSCREENGUI
-	g_touchscreengui->hide();
-#endif
+	if (g_settings->getBool("enable_touch"))
+		g_touchscreengui->hide();
```

Comment:
`enable_touch` can change anytime, so only hiding `g_touchscreengui` if `enable_touch` is true doesn't suffice.

For example, set `enable_touch = true` and put this into a mod:

```lua
minetest.register_on_joinplayer(function()
    minetest.after(1, function()
        minetest.settings:set("enable_touch", "false")
        minetest.request_shutdown()
    end)
end)
```

Result: The touchscreen controls stay visible on the shutdown screen.

There are probably many more problems like this with this PR. That's why I repeat my suggestion:
 1. only set `g_touchscreengui` to something other than null if `enable_touch` is true
 2. and then check for `g_touchscreengui` instead of `enable_touch`

We can then make in-game switching work properly using `SettingsChangedCallback`.

Reply:
Does that mean you agree to decide in the client launcher if we will have touchscreen support or not and then sticking to it?

Reply:
> Does that mean you agree to decide in the client launcher if we will have touchscreen support or not and then sticking to it?

Yes, I want "touchscreen controls enablement" to be decided here, in `clientlauncher.cpp`:

https://github.com/minetest/minetest/blob/32e492837cbf286aeb91b4b63ecf3c890c71a1bc/src/client/clientlauncher.cpp#L259-L272

This code runs each time you start a game from the mainmenu.

To make switching in-game possible (without going back to the mainmenu first), we can use a `SettingsChangedCallback`. This doesn't necessarily need to happen in this PR.